File: 1137 N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0: return 0
        elif n == 1: return 1        
        n0, n1, n2 = 0, 1, 1
        for i in range(3, n + 1):
            n0, n1, n2 = n1, n2, n0 + n1 + n2
        
        return n2

File: 1137 N-th Tribonacci Number.java
class Solution {
    public int tribonacci(int n) {
        if (n == 0) return 0;
        else if (n == 1) return 1;        
        
        int n0 = 0, n1 = 1, n2 = 1;
        for (int i = 3; i <= n; ++i) {
            int temp = n0 + n1 + n2;
            n0 = n1;
            n1 = n2;
            n2 = temp;
        }
        return n2;
    }
}